src/app/providers/datatables.provider.js

```diff
@@ -151,12 +191,26 @@ class DataTablesProvider {
             if (angular.element(tr).hasClass('detailedInfoRow'))
                 return;
 
+            angular.forEach(angular.element(`#${this.tableId} tbody tr`), function(trTmp, key){
+                if (angular.element(trTmp).hasClass('detailedInfoRow')) {
+                    angular.element(trTmp).remove();
+                } else {
+                    angular.element(trTmp).removeClass('shown');
+                }
+            });
+
             if (angular.element(tr).next().hasClass('detailedInfoRow')) {
                 angular.element(tr).next().remove();
                 tr.removeClass('shown');
             } else {
-                angular.element(row.node()).after(this.formatExtraInfo(index, row.node(), row.data()));
-                tr.addClass('shown');
+                let extraInfo = this.formatExtraInfo(index, row.node(), row.data());
+                angular.element(tr).injector().invoke(function($compile) {
```

Comment:
What we are doing here ? 
A single line comment would help any dev jumping into this.

Reply:
I have added the comment as follows:
```
let extraInfo = this.formatExtraInfo(EnvService, index, row.node(), row.data());
// Embedded "extraInfo" after digesting the directives within (They are displayed as is otherwise)
angular.element(tr).injector().invoke(function($compile) {
```